Makefile

```diff
@@ -1,37 +1,61 @@
-.ONESHELL:
 .PHONY:
 
 all: install
 
 ##### Variables ######
 COLOR := "\e[1;36m%s\e[0m\n"
 
+# Disable cgo by default.
+CGO_ENABLED ?= 0
+
 TEMPORAL_ROOT := temporal
 TCTL_ROOT := tctl
+CLI_ROOT := cli
+DOCKERIZE_ROOT := dockerize
 IMAGE_TAG ?= sha-$(shell git rev-parse --short HEAD)
 TEMPORAL_SHA := $(shell sh -c 'git submodule status -- temporal | cut -c2-40')
 TCTL_SHA := $(shell sh -c "git submodule status -- tctl | cut -c2-40")
 
 DOCKER ?= docker buildx
 BAKE := IMAGE_TAG=$(IMAGE_TAG) TEMPORAL_SHA=$(TEMPORAL_SHA) TCTL_SHA=$(TCTL_SHA) $(DOCKER) bake
+NATIVE_ARCH := $(shell go env GOARCH)
 
 ##### Scripts ######
 install: install-submodules
+clean:
+	rm -rf ./build
 
 update: update-submodules
 
 install-submodules:
-	@printf $(COLOR) "Installing temporal and tctl submodules..."
-	git submodule update --init $(TEMPORAL_ROOT) $(TCTL_ROOT)
+	@printf $(COLOR) "Installing submodules..."
+	git submodule update --init
 
 update-submodules:
 	@printf $(COLOR) "Updatinging temporal and tctl submodules..."
 	git submodule update --force --remote $(TEMPORAL_ROOT) $(TCTL_ROOT)
 
 ##### Docker #####
 
-build:
-	$(BAKE)
+# If you're new to Make, this is a pattern rule: https://www.gnu.org/software/make/manual/html_node/Pattern-Rules.html#Pattern-Rules
+# $* expands to the stem that matches the %, so when the target is amd64-bins $* expands to amd64
+%-bins:
+	mkdir -p build/$*
+	cd $(DOCKERIZE_ROOT) && CGO_ENABLED=$(CGO_ENABLED) GOOS=linux GOARCH=$* go build -o dockerize .
+	@cp $(DOCKERIZE_ROOT)/dockerize build/$*/
+	@GOOS=linux GOARCH=$* CGO_ENABLED=$(CGO_ENABLED) make -C $(TEMPORAL_ROOT) bins
+	@cp $(TEMPORAL_ROOT)/temporal-server build/$*/
+	@cp $(TEMPORAL_ROOT)/temporal-cassandra-tool build/$*/
+	@cp $(TEMPORAL_ROOT)/temporal-sql-tool build/$*/
+	@cp $(TEMPORAL_ROOT)/tdbg build/$*/
```

Comment:
I guess it doesn't have to be this PR, but I'd love it if we could switch the temporal makefile to use `go install` instead of `go build` and then this can just set GOBIN when it runs `make bins`

Reply:
I'll do that as followup work

Reply:
actually even better: `go install` everything and get rid of all the submodules. we just need to copy the `-tags` here, which is a little annoying, but we already copy `CGO_ENABLED`

Reply:
Note to future readers: we cannot use `go install` when cross-compiling: https://github.com/golang/go/issues/57485.

What we _should_ do later to resolve both this and the isolation issues is break this into a two step process like I've detailed elsewhere, but we need to stabilize this process before we perfect it.